### PR TITLE
Fixed CourseSummary subregions displaying as regions bug

### PR DESCRIPTION
**Pull request by jameslawesynergy**
Fixed CourseSummary subregions displaying as regions bug

### CODE REVIEW

File: src/Dfc.CourseDirectory.Web/Controllers/CourseSummaryController.cs [hunk 1]
```diff
@@ -118,45 +118,20 @@ public IActionResult Index(Guid? courseId, Guid? courseRunId)
...
-                var regions = GetRegions().RegionItems.Select(x => x.Id);
-                vm.Regions = FormattedRegionsByIds(allRegions, courseRun.Regions);
+                var allRegions = _courseService.GetRegions();
+
+                vm.Regions = allRegions.RegionItems
+                    .Select(r => new { r.Id, r.RegionName })
+                    .Concat(allRegions.RegionItems
+                        .SelectMany(r => r.SubRegion)
+                        .Select(r => new { r.Id, RegionName = r.SubRegionName }))
+                    .Where(r => allRegions.SubRegionsDataCleanse(courseRun.Regions.ToList()).Contains(r.Id))
```
Comment (timabell) [inline src/Dfc.CourseDirectory.Web/Controllers/CourseSummaryController.cs:130]: I think that `ToList` going to be run for every record. You might want to extract it to a variable or something.

File: src/Dfc.CourseDirectory.Web/Controllers/CourseSummaryController.cs [hunk 2]
```diff
@@ -118,45 +118,20 @@ public IActionResult Index(Guid? courseId, Guid? courseRunId)
...
 
-            if(courseRun.Regions != null)
+            if (courseRun.Regions != null)
             {
-                var allRegions = _courseService.GetRegions().RegionItems;
-                var regions = GetRegions().RegionItems.Select(x => x.Id);
-                vm.Regions = FormattedRegionsByIds(allRegions, courseRun.Regions);
+                var allRegions = _courseService.GetRegions();
+
+                vm.Regions = allRegions.RegionItems
```
Comment (timabell) [inline src/Dfc.CourseDirectory.Web/Controllers/CourseSummaryController.cs:125]: I'm finding this linq pretty hard to parse in terms of intent. Could you figure out a way to make the intent clearer?

**Review by timabell — APPROVED** [review]
[I would expect to see addition of test coverage for modifying legacy code.](https://timwise.co.uk/2020/07/09/approaches-to-refactoring-and-technical-debt/)

File: src/Dfc.CourseDirectory.Web/Controllers/CourseSummaryController.cs (hunk 1 above)
Comment (jameslawesynergy) [reply src/Dfc.CourseDirectory.Web/Controllers/CourseSummaryController.cs]: Good spot - fixed.

File: src/Dfc.CourseDirectory.Web/Controllers/CourseSummaryController.cs (hunk 2 above)
Comment (jameslawesynergy) [reply src/Dfc.CourseDirectory.Web/Controllers/CourseSummaryController.cs]: Updated as discussed.

**Comment by jameslawesynergy** [comment]
Cathie to confirm with Alix/Danny if this bug requires fixing or not.

**Comment by jameslawesynergy** [comment]
Discussed with Cathie:

re courses and regions v sub-regions having different displays, discussed with Danny, Alix and Clair - this is prior behaviour so they haven't changed anything.  UR would be best for this and we don't have this yet.  There could be some providers that have lots of sub-regions and we don't know how this would look.
tl:dr; please leave courses as-is for the mo
